sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -216,6 +250,20 @@ def plot(self, plot_method="contourf", ax=None, xlabel=None, ylabel=None, **kwar
         if ax is None:
             _, ax = plt.subplots()
 
+        estimator_name_to_display = (
+            estimator_name if estimator_name is not None else self.estimator_name
+        )
+        response_method_to_display = (
+            response_method if response_method is not None else self.response_method
+        )
+
+        if estimator_name_to_display or response_method_to_display:
+            title = (
+                f"{estimator_name_to_display} using "
+                f'"{response_method_to_display}" method'
```

Comment:
Again, I don't think this logic works, if only one of name or method is given, you don't want all the text.
I am also not sure about adding a title if we only have response method. Maybe @StefanieSenger can offer her opinion